packages/template-retail-react-app/app/pages/account/payments.jsx

```diff
@@ -92,40 +94,84 @@ CardPaymentForm.propTypes = {
 
 const AccountPaymentMethods = () => {
     const {formatMessage} = useIntl()
-    const {
-        isRegistered,
-        paymentInstruments,
-        addSavedPaymentInstrument,
-        removeSavedPaymentInstrument
-    } = useCustomer()
+    const customer = useCurrentCustomer()
+    const {isRegistered, paymentInstruments, isLoading} = customer
+    const addSavedPaymentInstrumentAction = useShopperCustomersMutation({
+        action: 'createCustomerPaymentInstrument'
+    })
+    const removeSavedPaymentInstrumentAction = useShopperCustomersMutation({
+        action: 'deleteCustomerPaymentInstrument'
+    })
+
     const [isEditing, setIsEditing] = useState(false)
     const toast = useToast()
     const form = useForm()
 
     const hasSavedPayments = paymentInstruments?.length > 0
-
     const submitForm = async (values) => {
+        const paymentInstrument = createCreditCardPaymentBodyFromForm(values)
+        const body = {
+            bankRoutingNumber: '',
+            giftCertificateCode: '',
+            ...paymentInstrument,
+            paymentCard: {
+                ...paymentInstrument.paymentCard,
+                securityCode: undefined
+            }
+        }
         try {
```

Comment:
This try...catch... block won't work because `mutate` is not async. Change the error handling to either:

1
```
try {
  await mutateAsync()  
} catch {

}

```

2
```
mutate({onError})
```

Reply:
Have changed the code to use mutateAsync